datalad/support/gitrepo.py

```diff
@@ -1001,6 +1006,45 @@ def _get_prefixed_commit_msg(msg):
         DATALAD_PREFIX = "[DATALAD]"
         return DATALAD_PREFIX if not msg else "%s %s" % (DATALAD_PREFIX, msg)
 
+    def configure_fake_dates(self):
+        """Configure repository to use fake dates.
+        """
+        lgr.debug("Enabling fake dates")
+        # FIXME: We should probably store this in .datalad/config instead of
+        # .git/config so that the setting is tracked, but we can't just change
+        # to where='dataset' below and then call self.commit because
+        # AnnexRepo.commit will fail if it's called before it sets
+        # self._batched.
+        self.config.set("datalad.dataset.fakedates", "true", where="local")
+
+    @property
+    def fake_dates_enabled(self):
+        """Is the repository configured to use fake dates?
+        """
+        if self._fake_dates_enabled is None:
+            self._fake_dates_enabled = \
+                self.config.getbool('datalad.dataset', 'fakedates', default=False)
+        return self._fake_dates_enabled
+
+    def add_fake_dates(self, env):
+        """Add fake dates to `env`.
+        """
+        last_date = self._git_custom_command(
+            None,
+            ["git", "for-each-ref", "--count=1",
+             "--sort=-committerdate", "--format=%(committerdate:unix)",
+             "refs/heads"])[0].strip()
+
+        seconds = int(last_date) if last_date else FAKE_DATE_ROOT
+        date = "{} +0000".format(seconds + 1)
+
+        lgr.debug("Setting date to {}".format(date))
+
+        if date:
+            env["GIT_AUTHOR_DATE"] = date
+            env["GIT_COMMITTER_DATE"] = date
+            env["GIT_ANNEX_VECTOR_CLOCK"] = str(seconds)
```

Comment:
Should it be up to nanosec even if only for consistency with other git-annex dates?
```shell
$> grep . **/MD5*
bd6/9c7/MD5E-s4--df8b712c4fe20a0df933819665770165.log:1522963990.372779603s 1 c09e667a-b759-4fa4-9142-a7614b214d9b
ff4/c57/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f.log:1112911996s 1 c09e667a-b759-4fa4-9142-a7614b214d9b
```

Reply:
Hmm, yeah, that's how getPOSIXTime from Data.Time.Clock.POSIX returns it, but since we're getting the time from git commit timestamps, we're limited to second resolution.  And git-annex's parsePOSIXTime explicitly handles the bare seconds format:

    {- Parses how POSIXTime shows itself: "1431286201.113452s"
     - Also handles the format with no fractional seconds. -}
    parsePOSIXTime :: String -> Maybe POSIXTime

Is your point just that you'd prefer us append 000000000?

Reply:
Yeah... Although now I am thinking "what for?". May be so that if someone to implement it's own parsing if those entries, but then second one would need to be supported anyways. So primarily for rare visual inspection of those logs may be? Anyways, probably doesn't matter and I am ok either way

Reply:
OK, then I'll stick with my preference of **not** appending the unnecessary zeros.